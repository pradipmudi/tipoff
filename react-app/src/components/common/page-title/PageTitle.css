.pageTitle{
    color:#333 !important;
}

.app-spinner {
    margin-left:1em
}