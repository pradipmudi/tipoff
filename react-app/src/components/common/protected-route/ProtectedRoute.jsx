import React from 'react';
import {Route, Redirect} from 'react-router-dom';
import { useSelector, shallowEqual } from 'react-redux';

import { Container } from 'bloomer';

import {selectCurrentUser} from '../../../redux/user/user-selectors';

export const ProtectedRoute = ({component: Component, ...rest}) => {
    const currentUser = useSelector(selectCurrentUser, shallowEqual);
    return (
        <Route 
            {...rest} 
            render={
                (props) => {
                    if (currentUser.authenticated) {
                        return (
                            <div>
                                <Container isFluid id="app-container">
                                    <div style={{padding:'1em 0em'}}>
                                        <Component {...props} />
                                    </div>
                                </Container>
                            </div>
                        )
                    } else {
                        return <Redirect exact to="/" />
                    }
                }
            }
        />
    )
}

export default ProtectedRoute;
