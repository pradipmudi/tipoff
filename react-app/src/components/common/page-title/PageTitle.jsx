import React from 'react';
import {Title, Icon} from 'bloomer';
import './PageTitle.css'

export default function  PageTitle  ({pageTitle, loading})  {
    
    document.title = 'Tipoff - '+pageTitle;

    return (<Title isSize={2} className="pageTitle">
        {pageTitle}
        {loading ? (<Icon className="fas fa-spin fa-spinner app-spinner" />) 
        : 
        (<React.Fragment></React.Fragment>)}
        
    </Title>
)};
