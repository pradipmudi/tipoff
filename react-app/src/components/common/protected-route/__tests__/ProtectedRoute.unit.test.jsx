import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import {Provider} from 'react-redux';
import store from '../../../../redux/store';

import {ProtectedRoute} from '../ProtectedRoute';
import PageTitle from '../../page-title/PageTitle';


describe("ProtectedRoute", () => {
    let props;

    beforeEach(() => {
        props = {
            component: PageTitle,
            currentUser : {
                authenticated : true,
                name: 'Test Test'
            }
        }
    })

    it("matches snapshot", () => {
        const snap = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <ProtectedRoute {...props} />
                </MemoryRouter>
            </Provider>
       ).toJSON();
        expect(snap).toMatchSnapshot();
    });

    it("renders correct component when user is authenticated", () => {
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={["/"]}>
                    <ProtectedRoute  {...props}/>
                </MemoryRouter>
            </Provider>
        );
        expect(wrapper.exists()).toEqual(true);
        expect(wrapper.find('PageTitle').length).toEqual(1);
    });

    it("redirects to `/` when not authenticated", () => {
        //un authenticate the user
        props.currentUser.authenticated = false;
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={["/"]}>
                    <ProtectedRoute  {...props}/>
                </MemoryRouter>
            </Provider>
        );
        expect(wrapper.find('Redirect').length).toEqual(1);
        expect(wrapper.find('Redirect').props().to).toEqual("/");
    });
});