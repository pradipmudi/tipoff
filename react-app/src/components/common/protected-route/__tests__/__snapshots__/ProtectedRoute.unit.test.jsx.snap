// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute matches snapshot 1`] = `
<div>
  <div
    className="container is-fluid"
    id="app-container"
  >
    <div
      style={
        Object {
          "padding": "20px",
        }
      }
    >
      <h1
        className="title is-2 pageTitle"
      />
    </div>
  </div>
</div>
`;
